duckduckgo/iOS
# Content Blocker improvements

Task/Issue URL: https://app.asana.com/0/414235014887631/1200341959926298
Tech Design URL: https://app.asana.com/0/414235014887631/1200361182088351
CC: 

**Description**:
Rewrite of Content Blocker Manager and supporting files to handle edge cases, improve thread safety and performance. See tech design for details.

**Steps to test this PR**:

Generic smoke tests.

Ensure tests are handling all expected and edge cases.

Look into:
- Content blocker effectiveness .
- Performance when cold starting.
- Performance when updating lists/toggling protection.
- Handling TDS and TempList updates (error cases, lost connections etc).

Test onboarding, to ensure https://app.asana.com/0/414235014887631/1200195915061851 glitch has been fixed.



**Device Testing**:


**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
